src/tests/test_pubkey.cpp

```diff
@@ -742,6 +745,72 @@ Test::Result PK_Key_Generation_Stability_Test::run_one_test(const std::string&,
    return result;
 }
 
+/**
+ * @brief Some general tests for minimal API sanity for signing/verification.
+ */
+class PK_API_Sign_Test : public Text_Based_Test {
+   public:
+      PK_API_Sign_Test() : Text_Based_Test("pubkey/api_sign.vec", "Algorithm,AlgoParams,SigParams", "Provider") {}
+
+   protected:
+      Test::Result run_one_test(const std::string&, const VarMap& vars) final {
+         const std::string algorithm = vars.get_req_str("Algorithm");
+         const std::string algo_params = vars.get_req_str("AlgoParams");
+         const std::string sig_params = vars.get_req_str("SigParams");
+         const std::string verify_params = vars.get_opt_str("VerifyParams", sig_params);
+         const std::string provider = vars.get_opt_str("Provider", "base");
+
+         std::ostringstream test_name;
+         test_name << "Sign/verify API tests " << algorithm;
+         if(!algo_params.empty()) {
+            test_name << '(' << algo_params << ')';
+         }
+         if(!sig_params.empty()) {
+            test_name << '/' << sig_params;
+         }
+         Test::Result result(test_name.str());
+
+         std::unique_ptr<Botan::Private_Key> privkey =
+            Botan::create_private_key(algorithm, Test::rng(), algo_params, provider);
+         if(!privkey) {
+            result.test_note(Botan::fmt(
+               "Skipping Sign/verify API tests for {}({}) with provider {}", algorithm, algo_params, provider));
+            return result;
+         }
+         std::unique_ptr<Botan::Public_Key> pubkey = Botan::X509::load_key(Botan::X509::BER_encode(*privkey));
+         result.confirm("Storing and loading public key works", pubkey != nullptr);
+
+         result.confirm("private key claims to support signatures",
+                        privkey->supports_operation(Botan::PublicKeyOperation::Signature));
+         result.confirm("public key claims to support signatures",
+                        pubkey->supports_operation(Botan::PublicKeyOperation::Signature));
+         result.test_gt("Public key length must be greater than 0", privkey->key_length(), 0);
+
+         std::unique_ptr<Botan::PK_Signer> signer = std::make_unique<Botan::PK_Signer>(
```

Comment:
For cases like this where the type is being named already in the expression that initializes the type, I think `auto` is a little easier.